src/terminal/input/terminalInput.hpp

```diff
@@ -87,7 +80,13 @@ namespace Microsoft::Console::VirtualTerminal
         bool _SearchWithModifier(const KeyEvent& keyEvent) const;
 
 
-    public:
+	public:

```

Comment:
nit: I'd prefer if the `public:` methods were listed first, before the `private` ones.

Reply:
@zadjii-msft  Should this guideline be added to the Coding Style guideline or a similar document?  I took a look at the repo and found a "[coding style](https://github.com/Microsoft/Terminal/blob/master/doc/STYLE.md)" document but it doesn't go into a lot of detail.   Is the plan to create a style guide that explicitly states the Dos/Don't for this repo? I'm sure you have this type of info documented internally and it would be great for external contributors.

Reply:
IIRC we're using some public C++ standard that I can't remember the name of but @adiviness definitely remembers the name of

Reply:
This one was referenced here quite a lot: http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines

Reply:
@mKay00 Thanks.  I found the [guideline](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#nl16-use-a-conventional-class-member-declaration-order)

Reply:
Yup, we attempt to follow the Windows style conventions as much as possible whenever we alter or write new code. I don't think the style guidelines are public but they defer a bunch of styling to the CppCoreGuidelines mentioned above so that would be a good place to start.

Reply:
The [ISO guideline](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#nl16-use-a-conventional-class-member-declaration-order) mentions to do this "when you have no constraints".

There is a constraint in one of the pulblic functions. `GetKeyMapping` return type is a `const _TermKeyMap*`. The struct `_TermKeyMap` is defined in the private section. Should the struct be declared public or should there be two public sections?

Reply:
Oh well that's weird. That struct should definitely remain private. `GetKeyMappingLength` and `GetKeyMapping` should probably be made private, since they're not used outside that class anyways.